source/magnifier/commands.py

```diff
@@ -0,0 +1,213 @@
+	log.info(f"Toggling filter for magnifier: {magnifier}")
+	if magnifierIsActiveVerify(
+		magnifier,
+		MagnifierAction.TOGGLE_FILTER,
+	):
+		filters = list(Filter)
+		idx = filters.index(magnifier.filterType)
+		magnifier.filterType = filters[(idx + 1) % len(filters)]
+		if magnifier.magnifierType == MagnifierType.FULLSCREEN:
+			magnifier._applyFilter()
+
+		# Save to config if option is enabled
+		if shouldSaveShortcutChanges():
+			setDefaultFilter(magnifier.filterType)
+
+		ui.message(
+			_(
+				# Translators: Message announced when changing the color filter with {filter} being the new color filter
+				"Color filter changed to {filter}",
+			).format(filter=magnifier.filterType.displayString),
+		)
+
+
+def toggleFullscreenMode():
+	"""Cycle through fullscreen focus modes (center, border, relative)."""
+	magnifier: Magnifier = getMagnifier()
+	if magnifierIsActiveVerify(
+		magnifier,
+		MagnifierAction.CHANGE_FULLSCREEN_MODE,
+	):
+		if magnifierIsFullscreenVerify(
+			magnifier,
+			MagnifierAction.CHANGE_FULLSCREEN_MODE,
+		):
+			modes = list(FullScreenMode)
+			currentMode = magnifier.fullscreenMode
+			idx = modes.index(currentMode)
+			newMode = modes[(idx + 1) % len(modes)]
+			log.info(f"Changing fullscreen mode from {currentMode} to {newMode}")
+			magnifier.fullscreenMode = newMode
+
+			# Save to config if option is enabled
+			if shouldSaveShortcutChanges():
+				setDefaultFullscreenMode(newMode)
+
+			ui.message(
+				_(
+					# Translators: Message announced when changing the fullscreen mode with {mode} being the new fullscreen mode
+					"Fullscreen mode changed to {mode}",
+				).format(mode=newMode.displayString),
+			)
+
+
+def startSpotlight():
+	magnifier: FullScreenMagnifier = getMagnifier()
+	if magnifierIsActiveVerify(
+		magnifier,
+		MagnifierAction.START_SPOTLIGHT,
+	):
+		if magnifierIsFullscreenVerify(
+			magnifier,
+			MagnifierAction.START_SPOTLIGHT,
+		):
+			log.info("trying to launch spotlight mode")
+			if magnifier._spotlightManager._spotlightIsActive:
+				log.info("found spotlight manager and it is active")
+				ui.message(
+					_(
+						# Translators: Message announced when trying to start spotlight mode while it's already active
+						"Spotlight mode is already active",
+					),
+				)
+			else:
+				log.info("no active spotlight manager found, starting new one")
+				magnifier._startSpotlight()
+				ui.message(
+					_(
+						# Translators: Message announced when spotlight mode is started
+						"Spotlight mode started",
+					),
+				)
+
+
+def magnifierIsActiveVerify(
+	magnifier: Magnifier,
+	action: MagnifierAction,
+) -> bool:
+	if magnifier and magnifier.isActive:
+		return True
+	else:
+		ui.message(
+			_(
+				# Translators: Message announced that the magnifier is not active
+				"Magnifier is not active at {action}",
+			).format(action=action.displayString),
+		)
+		return False
+
+
+def magnifierIsFullscreenVerify(
+	magnifier: Magnifier,
+	action: MagnifierAction,
```

Comment:
I think this could be change to this

```suggestion
	action: Literal[MagnifierAction.CHANGE_FULLSCREEN_MODE | MagnifierAction.START_SPOTLIGHT],
```

Reply:
should we check for it?